scheduler/src/cook/mesos/optimizer.clj

```diff
@@ -0,0 +1,134 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns cook.mesos.optimizer
+  (:require [cheshire.core :as cheshire]
+            [chime :refer [chime-at chime-ch]]
+            [clj-http.client :as http]
+            [clj-time.core :as t]
+            [clj-time.coerce :as tc]
+            [clj-time.periodic :as periodic]
+            [clojure.core.async :as async]
+            [clojure.tools.logging :as log]
+            [cook.util :refer [lazy-load-var]]
+            [cook.mesos.util :as util]
+            [datomic.api :as d :refer (q)]
+            [schema.core :as s]))
+
+(def PosNum
+  (s/both s/Num (s/pred pos? 'pos?)))
+
+(def PosInt
+  (s/both s/Int (s/pred pos? 'pos?)))
+
+(def NonNegInt
+  (s/both s/Int (s/pred #(<= 0 %) 'pos?)))
+
+(def TimePeriodMs NonNegInt)
+
+;; Elements missing:
+;;   1. Attributes (chip set, az, ..) -- have a combinitorial problem here
```

Comment:
What does "Elements missing" mean?

Reply:
Eh, it was me keeping track of keys I will probably want in the host info but don't need yet. I think it is worth just removing for now